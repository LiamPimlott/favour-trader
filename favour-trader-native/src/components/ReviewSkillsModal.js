import React, {Component} from 'react';
import {View, Text, Modal, } from 'react-native';
import { Button, List, ListItem } from 'react-native-elements'

class ReviewSkillsModal extends React.Component {
    state = {
        isOpen: null
    };

    componentDidMount() {
        const { isVisible } = this.props;
        this.setState({isOpen: isVisible})
    }

    render() {
        const { has, wants, toggle } = this.props;

        return (
            <Modal
                animationType="slide"
                transparent={false}
                visible={this.state.isOpen}
                onRequestClose={ () => {this.setState({modalVisible: false})}}>
                <View style={{marginTop: 22}}>
                    <View>
                        <Text>Has Skills</Text>
                        <List containerStyle={{marginBottom: 20}}>
                            {
                                has.map((item, i) => (
                                    <ListItem
                                        hideChevron={true}
                                        key={i}
                                        title={item.category.skill}
                                    />
                                ))
                            }
                        </List>
                        <Text>Want Skills</Text>
                        <List containerStyle={{marginBottom: 20}}>
                            {
                                wants.map((item, i) => (
                                    <ListItem
                                        key={i}
                                        hideChevron={true}
                                        title={item.category.skill}
                                    />
                                ))
                            }
                        </List>
                        <Button onPress={ () => {toggle()} } title={"Close"}>
                        </Button>
                    </View>
                </View>
            </Modal>
        )
    }
}

export default ReviewSkillsModal;