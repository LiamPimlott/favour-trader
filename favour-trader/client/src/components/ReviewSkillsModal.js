import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'antd'; 
import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

class ReviewSkillsModal extends Component {
    render() {
      const { isOpen, toggle, user } = this.props;
      return (
        <div>
        { 
          (user && user.address) ? (
            <div>
              <Modal isOpen={isOpen} toggle={toggle} backdrop={true}>
                <ModalHeader toggle={toggle}>{user.name.first + " " + user.name.last}</ModalHeader>
                <ModalBody>
                  <h6 style={{fontWeight: 'bold'}}>Skilled at:</h6>
                  <ul>
                    {
                      user.has.map( (skill) => {
                                      return (<li key={skill._id}> {skill.category.skill} </li>)
                                  }, this)
                    }
                  </ul> 
                  <h6 style={{fontWeight: 'bold'}}>Interested in:</h6>
                  <ul>
                    {
                      user.wants.map( (skill) => {
                                      return (<li key={skill._id}> {skill.category.skill} </li>)
                                  }, this)
                    }
                  </ul> 
                </ModalBody>
                <ModalFooter>
                  <Button type="primary" onClick={toggle}>
						<Link to={'/Profile/'+user._id}>
							View Profile
						</Link>
				  </Button>
                  <Button onClick={toggle}>OK</Button>
                </ModalFooter>
              </Modal>
            </div>
          ) : (
            ''
          )
        }
        </div>
      );
  }
}
export default ReviewSkillsModal;