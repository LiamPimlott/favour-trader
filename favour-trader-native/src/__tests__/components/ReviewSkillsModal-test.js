import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import moxios from "moxios";
import ReviewSkillsModal from '../../components/ReviewSkillsModal.js';
import toJson from 'enzyme-to-json'; //added this line

beforeEach(function () {
    const Adapter = require("enzyme-adapter-react-16");
    configure({adapter: new Adapter()});
    moxios.install();
});

describe('Testing ReviewSkillsModal component', () => {
    it('renders as expected', () => {
        const toggle = {bind:jest.fn()};
        const inVisible = false;
        const has = {map:jest.fn()};
        const wants = {map:jest.fn()};

        const wrapper = shallow(
            <ReviewSkillsModal toggle={toggle} inVisible={inVisible} has={has} wants={wants}/>
        );
        expect(toJson(wrapper)).toMatchSnapshot(); //edited this line
    });
});