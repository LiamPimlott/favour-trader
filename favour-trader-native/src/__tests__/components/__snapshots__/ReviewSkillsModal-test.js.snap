// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ReviewSkillsModal component renders as expected 1`] = `
<Component
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <View
    style={
      Object {
        "marginTop": 22,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Has Skills
      </Text>
      <List
        containerStyle={
          Object {
            "marginBottom": 20,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Want Skills
      </Text>
      <List
        containerStyle={
          Object {
            "marginBottom": 20,
          }
        }
      />
      <Button
        TouchableComponent={[Function]}
        buttonStyle={
          Object {
            "borderRadius": 3,
          }
        }
        clear={false}
        disabled={false}
        iconRight={false}
        loadingProps={
          Object {
            "color": "white",
            "size": "small",
          }
        }
        onPress={[Function]}
        title="Close"
      />
    </View>
  </View>
</Component>
`;
